src/backend/cdb/cdbmutate.c

```diff
@@ -2236,6 +2250,12 @@ shareinput_peekmot(ApplyShareInputContext *ctxt)
 	return linitial_int(ctxt->motStack);
 }
 
+static int
+shareinput_peekqds(ApplyShareInputContext *ctxt)
```

Comment:
In my opinion, meaning of qds term and its possible values is very unobvious.

Reply:
qds stands for query dispatcher slice

Reply:
> Maybe use enum?

done